java/ddlogapi/DDlogCommand.java

```diff
@@ -1,59 +1,47 @@
 package ddlogapi;
 
-import java.lang.reflect.*;
-
-public class DDlogCommand {
+/**
+ * Update to an input or output relation. 
+ *
+ * <p>
+ * DDlogCommand interface describes an update to a single record of a DDlog
+ * relation, which can be one of three kinds (1) insert a new record, (2) delete
+ * an existing record specifie by value, (3) delete an existing record specified
+ * by the record's primary key.
+ * </p>
+ *
+ * <p>
+ * This interface is used to represent both updates to DDlog input relations
+ * issued by the client and output updates generated by DDlog on transaction commit.
+ * </p>
+ *
+ * <p>
+ * The interface is parameterized by record type, which can be either DDlogRecord,
+ * used to represent any DDlog datatype in a generic way, or Object, when
```

Comment:
I don't understand this; value() already returns a value of type R.

Reply:
Sorry I don't understand this comment. `R` can be either `DDlogRecord` or `Object`.